functions/functionsv2/label_gce_instance/label_gce_instance_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package helloworld provides a set of Cloud Functions samples.
+package helloworld
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"google.golang.org/api/option"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+type FakeInstancesServer struct {
+	http.ServeMux
+	changedLabels map[string]string
+	instance      *computepb.Instance
+}
+
+func NewFakeInstancesServer(i *computepb.Instance) *FakeInstancesServer {
+	fake := &FakeInstancesServer{
+		instance:      i,
+		changedLabels: make(map[string]string),
+	}
+	fake.HandleFunc("/compute/v1/projects/PROJECT/zones/ZONE/instances/INSTANCE",
+		func(w http.ResponseWriter, r *http.Request) {
+			rbytes, err := json.Marshal(fake.instance)
```

Comment:
Since we are working with generated proto stubs we should use the protojson library: https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson

Reply:
Thanks, i did not know about the protojson library. done.